spring-integration-jmx/src/main/java/org/springframework/integration/monitor/IntegrationMBeanExporter.java

```diff
@@ -770,6 +784,10 @@ private void registerChannels() {
 				enabled = smartMatch(this.enabledStatsPatterns, name);
 				if (enabled != null) {
 					monitor.enableStats(enabled);
+					metrics.setFullStatsEnabled(enabled);
```

Comment:
Looks like this won't have a value if `!(monitor instanceof ConfigurableMetricsAware)`, although I think it won't hurt